upstairs/src/client.rs

```diff
@@ -857,6 +857,10 @@ impl DownstairsClient {
     ) -> EnqueueResult {
         match self.state {
             // We never send jobs if we're in certain inactive states
+            DsState::Connecting {
+                mode: ConnectionMode::New,
+                ..
+            } if self.cfg.read_only => EnqueueResult::Skip,
```

Comment:
Is this necessary?  I feel like it should bail out earlier, but am not 100% confident

Reply:
Okay, I've convinced myself that this makes sense.  You should put a comment saying that we can be connected in read-only mode with only a single Downstairs, so the other Downstairs should skip jobs in that case.

It seems like we _could_ also pass in `UpstairsState` to distinguish here?

Reply:
Comment added in f212546c